### DISCUSSION TITLE
Could ECO3D estimate inundation area

### DISCUSSION THREAD

**Discussion by AlexLiuxy** [changliao1025/eco3d#35] [General]
Is ECO3D able to estimate inundation area by keeping track of land and water in 3D fashion? If the answer is yes, is that possible to use current output variables to calculate, or we need to add some new variables?

**Comment by changliao1025** [comment]
The current version only keeps track of the total volume of water in each grid cell and assumes the river depth is infinite, the same treatment as the USGS PRMS.
But I may add the inundation feature in the next version to allow water to flood either on the floodplain or the neighboring cells.

**Reply by AlexLiuxy** [reply]
Thanks for reply. But if soil is inundated which means the saturated point is reached. I believe when dSoil_storage_moisture_max=1, that means the soil saturated point is reached and that region is inundated. I was wondering if that thought is true.

**Comment by changliao1025** [comment]
The dSoil_storage_moisture_max term is not a ratio, but instead a thickness term just like precipitation, then dSoil_storage_moisture_max  =1 does not mean it is saturated. 

// the max storage in preferential reservoir, unit: meter
        double dSoil_storage_preferential_max;
        double dSoil_storage_moisture_max;
